Source/Initialization/PlasmaInjector.cpp

```diff
@@ -79,14 +79,22 @@ PlasmaInjector::PlasmaInjector (int ispecies, const std::string& name)
     pp.query("radially_weighted", radially_weighted);
     AMREX_ALWAYS_ASSERT_WITH_MESSAGE(radially_weighted, "ERROR: Only radially_weighted=true is supported");
 
-    // parse plasma boundaries
-    xmin = std::numeric_limits<amrex::Real>::lowest();
-    ymin = std::numeric_limits<amrex::Real>::lowest();
-    zmin = std::numeric_limits<amrex::Real>::lowest();
-
-    xmax = std::numeric_limits<amrex::Real>::max();
-    ymax = std::numeric_limits<amrex::Real>::max();
-    zmax = std::numeric_limits<amrex::Real>::max();
+    // box boundaries
+    // FIXME: Only injects particles on coarsest grid, so changes are needed to allow injection to finer grid levels.
+    const Geometry& geom = WarpX::GetInstance().Geom(0);
+    const auto problo = geom.ProbLoArray();
+    const auto probhi = geom.ProbHiArray();
+    xmin = problo[0];
+    xmax = probhi[0];
+#ifndef WARPX_DIM_3D
+    zmin = problo[1];
+    zmax = probhi[1];
+#else
+    ymin = problo[1];
+    ymax = probhi[1];
+    zmin = problo[2];
+    zmax = probhi[2];
+#endif
```

Comment:
This will disable keep the some variables uniitialized in 2D, which can lead to segfaults and many `#if` needs down the road. For example, it also disables bounds check for particles in RZ ([docs](https://warpx.readthedocs.io/en/latest/developers/dimensionality.html)) or could lead to segfaults with new geometries.

I would suggest the following:
```suggestion
    ymin = problo[1];
    ymax = probhi[1];
    zmin = problo[2];
    zmax = probhi[2];

#ifdef WARPX_DIM_XZ
    ymin = std::numeric_limits<amrex::Real>::lowest();
    ymax = std::numeric_limits<amrex::Real>::max();
    zmin = problo[1];
    zmax = probhi[1];
#endif
```

Reply:
Oh true that. Thanks for pointing that out!
I think then that it makes more sense to write it this way:
```suggestion
#ifndef WARPX_DIM_3D
    ymin = std::numeric_limits<amrex::Real>::lowest();
    ymax = std::numeric_limits<amrex::Real>::max();
    zmin = problo[1];
    zmax = probhi[1];
#else
    ymin = problo[1];
    ymax = probhi[1];
    zmin = problo[2];
    zmax = probhi[2];
#endif
```